superset-frontend/src/dashboard/components/PropertiesModal/index.tsx

```diff
@@ -294,7 +296,12 @@ const PropertiesModal = ({
 
     // color scheme in json metadata has precedence over selection
     if (currentJsonMetadata?.length) {
-      const metadata = JSON.parse(currentJsonMetadata);
+      let metadata;
+      try {
+        metadata = JSON.parse(currentJsonMetadata);
+      } catch (error) {
+        addDangerToast(t('JSON metadata is invalid!'));
```

Comment:
@zhaoyongjie shouldn't this re-throw the error (or at least return from the function) if we don't have valid metadata here so we don't try to save an invalid payload?

Reply:
@etr2460 good point, let's try to make an invalid JSON to verify.

Reply:
@zhaoyongjie is there any update regarding a fix for this?

Reply:
@john-bodley @etr2460 add a new PR at [here](https://github.com/apache/superset/pull/21524).